lib/report.js

```diff
@@ -158,7 +160,162 @@ class ConciseReporter extends Reporter {
+        this.message += `${pad(6)}- ${el}\n`;
+      });
+      this.message += '  ...\n';
+      this.got = [];
+      this.expected = [];
+      console.log(this.message);
+    } else {
+      this.testNumeration.push(0);
+      this.lastIndex++;
+      this.calculateSpaces();
+      test.results.forEach(res => {
+        if (res.success) {
+          if (res.type === 'subtest') {
+            const t = res.test;
+            console.log(`${this.message}# Subtest: ${t.caption}`);
+            this.parseTestResults(t);
+            this.calculateSpaces();
+            const prefix = t.success ? 'ok' : 'not ok';
+            let message = `${this.message}${prefix} ${this.lastElement(0)}`;
+            message += ` - ${t.caption}`;
+            console.log(message);
+          } else {
+            this.listSuccess(res);
+          }
+        }
+      });
+      console.log(`${this.message}1..${this.lastElement(0)}`);
+      this.lastIndex--;
+      this.testNumeration.pop();
+    }
+    if (!this.lastIndex) this.message = '\n';
+  }
+
+  listSuccess(test) {
+    const testPrefix = test.success ? 'ok' : 'not ok';
+    console.log(`${this.message}${testPrefix} ${this.lastElement(1)}`);
+  }
+
+  listFailure(test, firstFailure) {
+    test.results.forEach(res => {
+      if (firstFailure) {
+        firstFailure = false;
+        this.message += '  ---\n';
+        if (res.message) this.message += `  message: '${res.message}'\n`;
+        this.message += '  severity: ';
+        this.message += res.severity ? `${res.severity}\n` : 'fail\n';
+        this.message += `${pad(2)}data:\n`;
+        this.message += `${pad(4)}got:\n`;
+      }
+      const data = [res.expected, res.actual].map(res => (
+        typeof res === 'function' || res instanceof Error ?
+          res.toString() :
+          res
+      ));
+      if (data[0] === undefined && data[1] === undefined) {
+        this.listFailure(res.test);
+      } else {
+        this.got.push(data[0]);
+        this.expected.push(data[1]);
+      }
+    });
+  }
+
+  record(test) {
+    let isSkipped = false;
+    if (!this.successful && !this.failed) console.log('TAP version 13');
+    if (test.success) {
+      this.successful++;
+    } else if (!test.todo) {
+      this.failed++;
+    } else {
+      this.skipped++;
+      isSkipped = true;
+    }
+    const testPrefix = test.success ? 'ok' : 'not ok';
+    const todoPrefix = test.todo ? ' #TODO' : '';
+    const skipPrefix = isSkipped ? ' #SKIP' : '';
+    let message = `\n${testPrefix} ${this.testNumeration[this.lastIndex]}`;
+    if (test.caption) {
+      message += ` -${skipPrefix || todoPrefix} ${test.caption}`;
+    }
+    console.log(message);
+    this.parseTestResults(test);
+  }
+
+  finish() {
+    const testCounter = this.skipped + this.failed + this.successful;
+    console.log(`1..${testCounter}`);
+    if (this.failed !== 0) {
+      let testHarness = `FAILED test`;
+      for (let i = 0; i < this.failed - 1; i++) {
```

Comment:
I'd prefer to have it this way since `this.failedTests.length` and `this.failed` are supposed to be equal.
```suggestion
      testHarness += ` ${this.failedTests.join(', ')}\n`;
```